pkg/config/config.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/amazon-eks-pod-identity-webhook/pkg/filesystem"
+	"k8s.io/klog/v2"
+	"sync"
+)
+
+type Config interface {
+	Get(namespace string, serviceAccount string) *ContainerCredentialsPatchConfig
+}
+
+type FileConfig struct {
+	containerCredentialsAudience string
+	containersCredentialsFullUri string
+
+	watcher              *filesystem.FileWatcher
+	identityConfigObject *IdentityConfigObject
+	cache                map[Identity]bool
+	mu                   sync.RWMutex // guards cache
+}
+
+type ContainerCredentialsPatchConfig struct {
+	Audience string
+	FullUri  string
+}
+
+func NewFileConfig(containersCredentialsAudience, containersCredentialsFullUri string) *FileConfig {
+	return &FileConfig{
+		containerCredentialsAudience: containersCredentialsAudience,
+		containersCredentialsFullUri: containersCredentialsFullUri,
+		identityConfigObject:         nil,
+		cache:                        make(map[Identity]bool),
+	}
+}
+
+func (f *FileConfig) StartWatcher(ctx context.Context, filePath string) error {
```

Comment:
Can you leave some comments about what this `ctx` represent? Does it contain a deadline of how fast the `Watch` action should be or how long we will be watching the file for?